api/query/search_test.go

```diff
@@ -16,6 +17,32 @@ import (
 	"google.golang.org/appengine/memcache"
 )
 
+func (mr *Mockreadable) MockSuccessfulRead(t *testing.T, ctrl *gomock.Controller, id string, v []byte) {
+	r := NewMockReader(ctrl)
+	mr.EXPECT().NewReader(id).Return(r, nil)
+	r.EXPECT().Read(gomock.Any()).DoAndReturn(func(p []byte) (int, error) {
+		assert.True(t, len(p) >= len(v))
+		for i := range v {
+			p[i] = v[i]
+		}
+		return len(v), nil
+	})
+	r.EXPECT().Read(gomock.Any()).Return(0, io.EOF)
+}
+
+func (mrw *MockreadWritable) MockSuccessfulRead(t *testing.T, ctrl *gomock.Controller, id string, v []byte) {
+	r := NewMockReader(ctrl)
+	mrw.EXPECT().NewReader(id).Return(r, nil)
+	r.EXPECT().Read(gomock.Any()).DoAndReturn(func(p []byte) (int, error) {
+		assert.True(t, len(p) >= len(v))
+		for i := range v {
+			p[i] = v[i]
+		}
+		return len(v), nil
+	})
+	r.EXPECT().Read(gomock.Any()).Return(0, io.EOF)
+}
```

Comment:
Can we simply make `store` (or other `Mockreadable`s) return a *fake* reader constructed by `bytes.NewReader` instead of mocking the reader from scratch? I don't think the assertions of buffer sizes in these mocks are related to the test cases. In fact, the documentation of `io.Reader` requires implementations to never return more than `len(p)` bytes, so these assertions are technically incorrect.